src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test.Common/LanguageServerTestBase.cs

```diff
@@ -63,6 +62,43 @@ protected static RazorCodeDocument CreateCodeDocument(string text, IReadOnlyList
             return codeDocument;
         }
 
+        internal static DocumentContextFactory CreateDocumentContextFactory(Uri documentPath, string sourceText)
+        {
+            var codeDocument = CreateCodeDocument(sourceText);
+            return CreateDocumentContextFactory(documentPath, codeDocument);
+        }
+
+        internal static DocumentContextFactory CreateDocumentContextFactory(
+            Uri documentPath,
+            RazorCodeDocument codeDocument,
+            bool documentFound = true,
+            int version = 1337)
+        {
+            var sourceTextChars = new char[codeDocument.Source.Length];
+            codeDocument.Source.CopyTo(0, sourceTextChars, 0, codeDocument.Source.Length);
+            var sourceText = SourceText.From(new string(sourceTextChars));
+
+            var snapshot = new Mock<DocumentSnapshot>(MockBehavior.Strict);
+            var outVal = documentFound ? codeDocument : null;
+            snapshot.Setup(s => s.TryGetGeneratedOutput(out outVal)).Returns(documentFound);
+            snapshot.Setup(s => s.GetTextAsync()).Returns(Task.FromResult(sourceText));
+            snapshot.Setup(s => s.GetGeneratedOutputAsync()).Returns(Task.FromResult(codeDocument));
+            snapshot.SetupGet(s => s.FilePath).Returns(documentPath.GetAbsoluteOrUNCPath());
+            snapshot.SetupGet(s => s.FileKind).Returns(codeDocument.GetFileKind());
+            
+            var documentContext = new DocumentContext(documentPath, snapshot.Object, version);
+            var documentContextFactory = new Mock<DocumentContextFactory>(MockBehavior.Strict);
```

Comment:
I thinkkkkkk there may exist a TestDocumentContextFactory already. It might be a private class